source/developers-guide/shopware-5-upgrade-guide-for-developers/index.md

```diff
@@ -17,14 +17,205 @@ including minor and bugfix releases, refer to the `UPGRADE.md` file found in you
+the last `replace` hook will then call the original implementation on `executeParent()`.
+ 
+### Filesystem abstraction layer
+
+As of Shopware v5.5, the use of direct file system access methods (like e.g. `fopen()`, `file()`, `is_readable()` and many more)
+is being discouraged. Instead, we rely on the library <a href="https://github.com/thephpleague/flysystem" target="_blank">FlySystem</a>
+as a generic file system abstraction layer. This allows a lot of flexibility in regards to where files are being stored.
+E.g. Documents, ESD and Sitemap files can now also served from S3 or Google Cloud. We added support for 
+Amazon Web Services and Google Cloud Platform for the moment but this layer will help us in the future to support more 
+cloud services. 
+
+Multiple instances of this abstraction classes are available to plugins. The services with the ids `shopware.filesystem.public`
+and `shopware.filesystem.private` are available to Shopware itself and to all plugins that are active. These services are
+used to access a kind of shared "folder" in which files can be stored that might be shared between Shopware and plugins.
+The service `shopware.filesystem.public.url_generator` can be used for generating a URL to a file in the public filesystem.
+
+The `shopware.filesystem.public` file system is intended for files that need to be accessible to clients directly, e.g. 
+images, product manuals etc. while `shopware.filesystem.private` is for files that need to be available throughout Shopware
+(e.g. invoice pdfs, ESDs like software), but are not to be accessed by a browser directly without any authentication.
+
+This is a simple example of how a file can be written using the new system:
+
+```php
+$fileName = 'HelloWorld.txt';
+$contents = 'Uploaded by Shopware using FlySystem';
+
+$filesystem = $this->container->get('shopware.filesystem.private');
+if ($filesystem->has($fileName)) {
+    $filesystem->delete($fileName);
+}
+
+$filesystem->write($file, $contents);
+```
+
+Plugins can access their own filesystems which are instantiated and available automatically:
+    * `plugin_name.filesystem.public`
+    * `plugin_name.filesystem.private`
+
+### Sitemap
+
+To support more than 50.000 URLs that are allowed in one `sitemap.xml` file, Shopware now adds a `shopware_index.xml`,
+which itself contains links to one or more `sitemap.xml`. The sitemap files can be created by cronjob, live or using
+the command `sw:generate:sitemap`.
+
+If you need to add some links to the sitemap, you can simply create a service implementing the interface
+`\Shopware\Bundle\SitemapBundle\UrlProviderInterface` and give it the DIC tag `sitemap_url_provider`. The sitemap exporter
+will collect your service using the tag and export the URLs it provides along with the others.
+
+### Dynamic cache invalidation
+
+interface `Shopware\Components\HttpCache\CacheTimeServiceInterface`, tag `invalidation_date_provider`
```

Comment:
nothing more to tell about that? :disappointed:

Reply:
I thought the product name is "elasticsearch", i have been fooled!

Reply:
Shit, forgot to write the text around that :D